File: search_techverse.py
import os

directory = "src"
output_file = "search_results.txt"
keyword = "techverse"

results = []

if os.path.exists(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.jsx', '.js', '.css', '.html')):
                path = os.path.join(root, file)
                with open(path, "r", encoding="utf-8", errors="ignore") as f:
                    content = f.read()
                if keyword in content.lower():
                    results.append(f"File: {path}")
                    for line_no, line in enumerate(content.split('\n'), 1):
                        if keyword in line.lower():
                            results.append(f"  {line_no}: {line.strip()}")
                    results.append("")
else:
    results.append("Directory not found")

with open(output_file, "w", encoding="utf-8") as f:
    f.write("\n".join(results))
print("Done searching")

File: search_techverse.js
const fs = require('fs');
const path = require('path');

const directory = path.join(__dirname, 'src');
const indexHtml = path.join(__dirname, 'index.html');
const keyword = 'techverse';

const results = [];

function searchFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    if (content.toLowerCase().includes(keyword)) {
      const lines = content.split('\n');
      results.push(`File: ${path.relative(__dirname, filePath)}`);
      lines.forEach((line, idx) => {
        if (line.toLowerCase().includes(keyword)) {
          results.push(`  ${idx + 1}: ${line.trim()}`);
        }
      });
      results.push('');
    }
  } catch (err) {
    console.error(`Error reading ${filePath}:`, err.message);
  }
}

function walkDir(dir) {
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat && stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git') {
        walkDir(filePath);
      }
    } else {
      if (filePath.endsWith('.jsx') || filePath.endsWith('.js') || filePath.endsWith('.css') || filePath.endsWith('.json')) {
        searchFile(filePath);
      }
    }
  });
}

if (fs.existsSync(directory)) {
  walkDir(directory);
}
if (fs.existsSync(indexHtml)) {
  searchFile(indexHtml);
}

fs.writeFileSync(path.join(__dirname, 'search_results_node.txt'), results.join('\n'));
console.log('Search complete. Output written to search_results_node.txt');
